### ISSUE TITLE
ChartFill doesn't seem to work - or I'm just not understanding it

### ISSUE THREAD

**Issue by akac** [danielgindi/Charts#2727]
I'm using a filled line chart. I want to have a simple gradient from one color to another. All I'm seeing is a single color, no matter if I use a gradient or not. 

Here is my code chunks :)

`	fileprivate let set_checkin = LineChartDataSet(values: [ChartDataEntry()], label: "Checkins")
`

Yes - its ugly orange to green - its for testing just to see if it works:

`		let checkinColors = [UIColor.orange, UIColor.green]
		set_checkin.fillAlpha = 1.0
		set_checkin.fill = Fill(linearGradient: CGGradient.gradient(withUIColors: checkinColors), angle: 90.0)
		set_checkin.drawFilledEnabled = true
`
And here is my gradient extension - I've tried multiple ways to create Gradients - but I just left this one.

`
 extension CGGradient {	
	class func gradient(withUIColors colors: [UIColor]) -> CGGradient {
		var colorFloats = [CGFloat]()
		
		for color in colors {
			var rSelf:CGFloat = 0, gSelf:CGFloat = 0, bSelf:CGFloat = 0, alpha:CGFloat = 0
			color.getRed(&rSelf, green: &gSelf, blue: &bSelf, alpha: &alpha)
			colorFloats.append(rSelf)
			colorFloats.append(gSelf)
			colorFloats.append(bSelf)
			colorFloats.append(alpha)
		}
		
		let colorSpace = CGColorSpaceCreateDeviceRGB()
		return CGGradient(colorSpace: colorSpace, colorComponents: colorFloats, locations: nil, count: colors.count)!
	}
}`

But all I get is a solid color - the first one in the gradient.

**Comment by liuxuan30** [comment]
have you looked into ChartsDemo `LineChart1ViewController`? it provides a demo:
```objective-c
        NSArray *gradientColors = @[
                                    (id)[ChartColorTemplates colorFromString:@"#00ff0000"].CGColor,
                                    (id)[ChartColorTemplates colorFromString:@"#ffff0000"].CGColor
                                    ];
        CGGradientRef gradient = CGGradientCreateWithColors(nil, (CFArrayRef)gradientColors, nil);

        set1.fillAlpha = 1.f;
        set1.fill = [ChartFill fillWithLinearGradient:gradient angle:90.f];
        set1.drawFilledEnabled = YES;

        CGGradientRelease(gradient);
```

**Comment by akac** [comment]
Yes, I've been using the demo. What I figured out is that the gradient starts from the very top of the chart down and my values weren't going that high.

**Comment by liuxuan30** [comment]
better ask this on SO, I am not quite understand but the demo seems working. Sorry can't offer more help
